tests/test_task.py

```diff
@@ -1,196 +1,462 @@
-    assert res_nml['cable']['spinup'] is False
-    assert res_nml['cable']['some_setting'] is True
-
-    assert res_nml['cable']['filename']['foo'] == 123, "assert existing derived types are preserved"
-    assert res_nml['cable']['bar'] == 123, "assert existing top-level parameters are preserved"
+    mock_subprocess = MockSubprocessWrapper()
+    task = get_mock_task(subprocess_handler=mock_subprocess)
+    task_dir = MOCK_CWD / internal.SITE_TASKS_DIR / task.get_task_name()
+    task_dir.mkdir(parents=True)
+    exe_path = task_dir / internal.CABLE_EXE
+    exe_path.touch()
+    nml_path = task_dir / internal.CABLE_NML
+    nml_path.touch()
 
+    stdout_file = task_dir / internal.CABLE_STDOUT_FILENAME
 
-def test_setup_task():
-    """Tests for `setup_task()`."""
+    # Success case: run CABLE executable in subprocess
+    task.run_cable()
+    assert f"{exe_path} {nml_path}" in mock_subprocess.commands
+    assert stdout_file.exists()
 
-    # Success case: test branch specific settings are patched into task namelist file
-    task = setup_mock_task()
-    task_dir = Path(TMP_DIR, internal.SITE_TASKS_DIR, task.get_task_name())
+    # Success case: test non-verbose output
+    with contextlib.redirect_stdout(io.StringIO()) as buf:
+        task.run_cable()
+    assert not buf.getvalue()
 
-    setup_mock_namelists_directory()
-    setup_fluxnet_directory_tree([task], root_dir=TMP_DIR)
-    do_mock_checkout_and_build()
+    # Success case: test verbose output
+    with contextlib.redirect_stdout(io.StringIO()) as buf:
+        task.run_cable(verbose=True)
+    assert not buf.getvalue()
 
-    task.setup_task(root_dir=TMP_DIR)
+    # Failure case: raise CableError on subprocess non-zero exit code
+    mock_subprocess.error_on_call = True
+    with pytest.raises(CableError):
+        task.run_cable()
+
+
+def test_add_provenance_info():
+    """Tests for `add_provenance_info()`."""
+
+    mock_subprocess = MockSubprocessWrapper()
+    task = get_mock_task(subprocess_handler=mock_subprocess)
+    task_dir = MOCK_CWD / internal.SITE_TASKS_DIR / task.get_task_name()
+    task_dir.mkdir(parents=True)
+    site_output_dir = MOCK_CWD / internal.SITE_OUTPUT_DIR
+    site_output_dir.mkdir()
+
+    # Create mock namelist file in task directory:
+    f90nml.write(
+        {"cable": {"filename": {"met": "/path/to/met/file", "foo": 123}, "bar": True}},
+        task_dir / internal.CABLE_NML,
+    )
+
+    # Create mock netcdf output file as if CABLE had just been run:
+    nc_output_path = site_output_dir / task.get_output_filename()
+    netCDF4.Dataset(nc_output_path, "w")
+
+    # Success case: add global attributes to netcdf file
+    task.add_provenance_info()
+    with netCDF4.Dataset(str(nc_output_path), "r") as nc_output:
+        atts = vars(nc_output)
+        assert atts["cable_branch"] == "mock standard output"
+        assert atts["svn_revision_number"] == "mock standard output"
```

Comment:
```suggestion
        assert atts["cable_branch"] == mock_subprocess.stdout
        assert atts["svn_revision_number"] == mock_subprocess.stdout
```

Reply:
Fixed in [95f1871](https://github.com/CABLE-LSM/benchcab/pull/99/commits/95f187105c25613915dc49b0185b4aebefff0f42)